MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -122,15 +123,14 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
     
     func notifyDidReroute(route: Route) {
         routePageViewController.notifyDidReRoute()
-        mapView.annotate([route], clearMap: true)
+        mapView.addArrow(routeController.routeProgress)
+        mapView.annotate(route)
```

Comment:
Let’s fix `annotate(_:)` the same way you’ve fixed `addArrow(_:)`: instead of blowing away the source and layers and adding them back in, just change the source’s shape. That’ll fix any flashing that occurs during rerouting and also reduce the amount of work MGLMapView has to do every time the route changes.